Storage/SQL/SQLiteHistory.swift

```diff
@@ -259,17 +299,22 @@ extension SQLiteHistory: BrowserHistory {
         return site
     }
 
-    private func getFilteredSitesWithLimit(limit: Int, whereURLContains filter: String?, orderBy: String, includeIcon: Bool) -> Deferred<Result<Cursor<Site>>> {
+    private func getFilteredSitesWithLimit(limit: Int,
+            whereURLContains filter: String? = nil,
+            groupClause: String = "GROUP BY \(TableHistory).id ",
+            orderBy: String = "ORDER BY visitDate DESC ",
+            whereData: String = "",
+            includeIcon: Bool = true) -> Deferred<Result<Cursor<Site>>> {
         let args: Args?
         let whereClause: String
         if let filter = filter {
             args = ["%\(filter)%", "%\(filter)%"]
 
             // No deleted item has a URL, so there is no need to explicitly add that here.
-            whereClause = " WHERE ((\(TableHistory).url LIKE ?) OR (\(TableHistory).title LIKE ?)) "
+            whereClause = " WHERE ((\(TableHistory).url LIKE ?) OR (\(TableHistory).title LIKE ?)) \(whereData)"
```

Comment:
Perhaps make `whereData` optional and include an explicit `AND` here? Requiring the caller to know that an `AND` is necessary seems tricky.